### PR TITLE
PP-973: PTL is not reverting the managers and operators in setUp

### PR DESCRIPTION
**Pull request by borlesanket**
[manager_unset.txt](https://github.com/PBSPro/pbspro/files/1984842/manager_unset.txt)

#### Bug/feature Description
* PTL setup does not revert users other than current user added to managers list.
* [PP-973](https://pbspro.atlassian.net/browse/PP-973)

#### Affected Platform(s)
* All platforms

#### Cause / Analysis / Design
* Existing PTL setup only removes current user from managers list, it does not care about the users other than current user.  

#### Solution Description
* Unset the managers instead of removing only current user from managers list inside PTL setup.

#### Testing logs/output
* *Please attach your test log output from running the test you added (or from existing tests that cover your changes)*

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [x] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: test/fw/ptl/utils/pbs_testsuite.py [hunk 1]
```diff
@@ -855,9 +855,9 @@ def revert_server(self, server, force=False):
         server_stat = server.status(SERVER)[0]
         current_user = pwd.getpwuid(os.getuid())[0]
         try:
-            # remove current user's entry from managers list (if any)
-            a = {ATTR_managers: (DECR, current_user + '@*')}
-            server.manager(MGR_CMD_SET, SERVER, a, sudo=True)
+            # Unset managers list
+            a = ['managers']
```
Comment (hirenvadalia) [inline test/fw/ptl/utils/pbs_testsuite.py]: Why do we need list? I think just 'managers' will be sufficient, no?

File: test/fw/ptl/utils/pbs_testsuite.py [hunk 2]
```diff
@@ -855,9 +855,9 @@ def revert_server(self, server, force=False):
...
+            server.manager(MGR_CMD_UNSET, SERVER, a, sudo=True)
```
Comment (hirenvadalia) [inline test/fw/ptl/utils/pbs_testsuite.py]: Can we add expect=True here as well as below server.manager()?

File: test/tests/selftest/pbs_managers_operators_ext.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    """
+    def test_managers_unset_setup(self):
+        """
+        Additional managers users, except current user should get unset
+        after test setUp run
+        """
+        runas = ROOT_USER
+        current_usr = pwd.getpwuid(os.getuid())[0]
+        a = {ATTR_managers: current_usr + '@*'}
```
Comment (bhroam) [inline test/tests/selftest/pbs_managers_operators_ext.py]: You build this dictionary twice.  Once here and once at the end of the test.  I'd create it once here and use it in both places.

File: test/fw/ptl/utils/pbs_testsuite.py [hunk 4]
```diff
@@ -855,9 +855,9 @@ def revert_server(self, server, force=False):
         server_stat = server.status(SERVER)[0]
         current_user = pwd.getpwuid(os.getuid())[0]
         try:
-            # remove current user's entry from managers list (if any)
-            a = {ATTR_managers: (DECR, current_user + '@*')}
-            server.manager(MGR_CMD_SET, SERVER, a, sudo=True)
+            # Unset managers list
+            server.manager(MGR_CMD_UNSET, SERVER, 'managers', sudo=True,
+                           expect=True)
```
Comment (bhroam) [inline test/fw/ptl/utils/pbs_testsuite.py]: Is it wise to except pass here?  If the manager fails, then you won't have unset the attribute.  Things won't be completely back to defaults at the beginning of the next test.  Maybe at least log something.

File: test/fw/ptl/utils/pbs_testsuite.py (hunk 2 above)
Comment (bhroam) [reply test/fw/ptl/utils/pbs_testsuite.py]: I suspect it has to do with the fact that you're incrementing two things at once.  I suspect if you broke it out into two manager calls, expect=True would work.  I'd call this a PTL issue.  Of course I'm not exactly sure what PTL should do with it.  You're adding two elements to a string array.  I guess PTL could break it on the comma and check if both are in the string.

Comment (hirenvadalia) [reply test/fw/ptl/utils/pbs_testsuite.py]: Yes this is expect issue... @borlesanket Can you please file an issue for this?

Comment (borlesanket) [reply test/fw/ptl/utils/pbs_testsuite.py]: I have logged PP-1277 for this issue.

File: test/fw/ptl/utils/pbs_testsuite.py [hunk 5]
```diff
@@ -858,8 +858,8 @@ def revert_server(self, server, force=False):
             # Unset managers list
             server.manager(MGR_CMD_UNSET, SERVER, 'managers', sudo=True,
                            expect=True)
-        except:
-            pass
+        except PbsManagerError, e:
```
Comment (bhroam) [inline test/fw/ptl/utils/pbs_testsuite.py]: Use the newer python exception form of using as instead of comma.

**Comment by borlesanket** [comment]
I have rebased this branch and its ready for merge.

**Pull request merged**